src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,30 +1,61 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+type Props = {
+  onFilterByQuery: (event: string) => void;
+  onFilterBy: (event: string) => void;
+  query: string;
+};
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+export const TodoFilter: React.FC<Props> = ({
+  onFilterByQuery = () => {},
+  onFilterBy = () => {},
+  query,
+}) => {
+  const handleChoose = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    onFilterBy(event.target.value);
+  };
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
-    </p>
-  </form>
-);
+  const handleSearchTodos = (event: React.ChangeEvent<HTMLInputElement>) => {
+    onFilterByQuery(event.target.value);
+  };
+
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select data-cy="statusSelect" onChange={handleChoose}>
+            <option value="all">All</option>
+            <option value="active">Active</option>
+            <option value="completed">Completed</option>
```

Comment:
When you have the enum you can create an array from it using `Object.values` and then map through it to generate options